parlai/core/worlds.py

```diff
@@ -1424,12 +1422,14 @@ def getID(self):
         """
         return self.inner_world.getID()
 
-    @lru_cache(maxsize=1)
     def num_examples(self):
         """
         Return the number of examples.
         """
-        return self.inner_world.num_examples()
+        if hasattr(self, '_num_examples'):
```

Comment:
i definitely prefer this logic anyway, the `lru_cache` seemed a little much for its purpose